app/validators/company.js

```diff
@@ -0,0 +1,70 @@
+const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
+const companyRepository = require('../repositories/company');
+const error = require('../services/error');
+
+class CompanyValidator {
+  async validateCreate(company) {
+    if (company.id !== undefined) {
+      error.throwValidationError('Invalid company format.');
+    }
+
+    const existingCompany = await companyRepository.getByPropsNonParanoid({
+      name: company.name,
+    });
+    if (existingCompany) {
+      error.throwValidationError('Company already exists.');
+    }
+
+    const {
+      business,
+      ...rest
+    } = company;
+    if (business !== undefined) {
+      const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({
+        name: business,
+      });
+      if (!fieldOfActivity) {
+        error.throwNotFoundError('Field of activity not found.');
+      }
+    } else {
+      error.throwValidationError('Invalid company format.');
+    }
+  }
+
+  async validateUpdate(id, data) {
+    if (data.id !== undefined) {
+      error.throwValidationError('You can not change the id.');
+    }
+
+    if (data.name !== undefined) {
+      error.throwValidationError('You can not change the company name.');
+    }
+
+    const existingCompany = await companyRepository.get(id);
+    if (!existingCompany) {
+      error.throwNotFoundError('Company not found.');
+    }
+
+    const {
```

Comment:
The same code can be found in the method above. This is not ok, the code is not DRY (Don't Repeat Yourself). Please move it in a method and call it in both places.